### PR TITLE
[MCC-402671] filter depricated security questions

### PR DESCRIPTION
**Pull request by agodel**
- Updates sample app with new babbage podspec
- Upgrades swift version to 4
- Fixes some crashes cause by swift update
- Consumes new structure of Security Question response and filters out deprecated questions

@geabraham

### CODE REVIEW

File: AppConnectSwift/SecurityQuestionViewController.swift [hunk 1]
```diff
@@ -8,6 +8,10 @@ class SecurityQuestionViewController: UIViewController, UITableViewDelegate, UIT
     var createAccountViewController = CreateAccountViewController()
     var userEmail : String!
     var userPassword : String!
+    let securityQuestionKey = "user_security_questions"
+    let depricatedKey = "deprecated"
```
Comment (ncarroll-mdsol) [inline AppConnectSwift/SecurityQuestionViewController.swift]: s/depricated/deprecated

File: AppConnectSwift/SecurityQuestionViewController.swift [hunk 2]
```diff
@@ -8,6 +8,10 @@ class SecurityQuestionViewController: UIViewController, UITableViewDelegate, UIT
...
+    let idKey = "id"
+    let questionKey = "name"
```
Comment (ncarroll-mdsol) [inline AppConnectSwift/SecurityQuestionViewController.swift]: Nitpick: these should probably be static to be consistent with our usage elsewhere

File: Podfile [hunk 3]
```diff
@@ -3,7 +3,7 @@ source 'https://github.com/CocoaPods/Specs.git'
 use_frameworks!
 
 pod 'IQKeyboardManagerSwift', '~>4.0'
-pod 'Babbage', podspec: 'AppConnectSwift/Babbage.podspec'
+pod 'Babbage', podspec: 'AppConnectSwift/Babbage.podspec.json'
```
Comment (ncarroll-mdsol) [inline Podfile]: I think we should still use a `.podspec` file rather than a `podspec.json`. It's more standard and is what we use elsewhere.

**Review by ncarroll-mdsol — CHANGES_REQUESTED** [review]

**Comment by agodel** [comment]
@ncarroll-mdsol addressed

**Review by ncarroll-mdsol — APPROVED** [review]

**Pull request merged**